File: src/2025-02-23-deepclaude-try/cpp/exercises/practice/sublist/sublist.cpp
#include "sublist.h"

namespace sublist {

namespace {
    bool isSublist(const std::vector<int>& sub, const std::vector<int>& super) {
        if (sub.empty()) {
            return true;
        }
        if (super.size() < sub.size()) {
            return false;
        }
        for (size_t i = 0; i <= super.size() - sub.size(); ++i) {
            bool match = true;
            for (size_t j = 0; j < sub.size(); ++j) {
                if (super[i + j] != sub[j]) {
                    match = false;
                    break;
                }
            }
            if (match) {
                return true;
            }
        }
        return false;
    }
}  // namespace

List_comparison sublist(const std::vector<int>& a, const std::vector<int>& b) {
    if (a == b) {
        return List_comparison::equal;
    }
    if (isSublist(a, b)) {
        return List_comparison::sublist;
    }
    if (isSublist(b, a)) {
        return List_comparison::superlist;
    }
    return List_comparison::unequal;
}

}  // namespace sublist

File: src/2025-02-23-deepclaude-try/go/exercises/practice/sublist/sublist.go
package sublist

// Relation type is defined in relations.go file.

func Sublist(l1, l2 []int) Relation {
	if areEqual(l1, l2) {
		return RelationEqual
	}
	if isSublist(l2, l1) {
		return RelationSublist
	}
	if isSublist(l1, l2) {
		return RelationSuperlist
	}
	return RelationUnequal
}

func areEqual(a, b []int) bool {
	if len(a) != len(b) {
		return false
	}
	for i, v := range a {
		if v != b[i] {
			return false
		}
	}
	return true
}

func isSublist(super, sub []int) bool {
	if len(sub) == 0 {
		return true
	}
	if len(super) < len(sub) {
		return false
	}
	for i := 0; i <= len(super)-len(sub); i++ {
		match := true
		for j := 0; j < len(sub); j++ {
			if super[i+j] != sub[j] {
				match = false
				break
			}
		}
		if match {
			return true
		}
	}
	return false
}
